lib/CMakeLists.txt

```diff
@@ -1,25 +1,32 @@
-add_library(prometheus-cpp
+set(PROMETHEUS_SOURCES 
```

Comment:
Instead of dynamically adding sources I'd like to see the lib split into two like #112 tries to do. Maybe we also should separate the files into separate directories. @jupp0r what do you think?

Reply:
I considered grouping it in a core library a separate change, as work in #112 was already underway. I can do that for CMake, however, I'm not familiar with the other build tool.

Reply:
For us to integrate properly into our project, the options to disable building CIVETWEB and testing is important, as we do not use it and it gives us troubles building the project.

Reply:
I think it would be nicer to be able to choose between a core library and one that comes with an embedded http server. My main concern is that an option will cause problems with distro packaging because only one option combination can ever be selected for a binary package. The core library option would allow us to distribute two packages so consumers can still use whatever option they want without building from source.

Reply:
So in practice, you prefer the repository to be split into two parts, each one with a separate CMakeLists.txt and the core one is included as an external project in the one with the webserver? Otherwise running cmake could result in unresolved dependencies (as would be for us).

Do you prefer a directory layout for this? "core" and "webserver" directories?

Reply:
I filed #123 and #124. Once they are merged I'll add options to select what components to build.